### ISSUE TITLE
THANK YOU! (and a question)

### ISSUE THREAD

**Issue by millercentral** [davecpearce/homeassistant-integrations#1]
Hello. I have been looking for a way to control my Smart Cocoon fans in home assistant and went so far as to log the REST APIs with Charles Proxy, but then I discovered your repository.

I was able to setup your custom component in my HA instance and it correctly found and can control the two Smart Cocoon fans I have in the house. Hurray!

The only issue I have found so far (which is not a deal breaker for me), is that it does not appear to reflect fan status changes that happen outside of Home Assistant (ie, through the SC app, or via Auto/Eco modes). Is it supposed to be able to do that, and if so, and suggestions on what I can do to debug?

Thank you for building this! I really appreciate it.

**Comment by davecpearce** [comment]
Hi there 

Thank you for reaching out. Unfortunately they do not coexist very well at this time as the login token used for HA is invalid when you connect to the app from what I found. There is an MQTT event we can subscribe to but this was getting very complicated and for me personally I did not need to use the SC app. I handle controls and automations within HA and Node Red. 

The MQTT logic is in the code but currently disabled. 

If you have the experience to take a look at the HA integration code and the Python API library and can make improvements, I would appreciate the help.

**Comment by millercentral** [comment]
The SC app is hot garbage, so I'm happy to stop using that, but it would be nice to know when the fan comes on because of eco/auto mode triggering it. Maybe this already works if I don't mess with the SC app.

I'm no python dev (partly why I was so excited to find this), but I'll poke around and experiment and let you know if I come up with anything interesting. This integration is going to make my hvac automations so much more interesting, thanks again.

**Comment by davecpearce** [comment]
I personally do not use the auto/eco feature myself in the SC app I added my own automation integrating with Ecobee in my case. 

If you don't log into the app, the MQTT events could work to pick up eco/auto events from the SC fan side to change the state in HA. 

Unfortunately I don't have the time to look into this right now. 

Hopefully you still get some value out of the integration.

**Comment by timnovinger** [comment]
@millercentral I unfortunately haven't been as lucky (although I'm using an outdated version of HA at the moment so I suppose it's possible that's the culprit) enough to get it working. I have the integration loaded and configured with my user/pass, but it's bombing during start up while trying to add the fan entities it seems.

Wondering if I did something wrong? Would you be willing to share what steps you took to get it working on your setup?

I also would love to avoid using the hot garbage that is their iOS app...

**Comment by millercentral** [comment]
So I realized after posting this that there is a different repro in dave's github that has a version that works with hacs: https://github.com/davecpearce/hacs_smartcocoon  -- I installed this and it is working well for me. It doesn't update state, but I've stopped using auto modes or the Smart Cocoon app so that isn't an issue. I've found its a bit more reliable to sperate on/off from setting fan speed, but overall it is working well for the two fans I have.

**Comment by davecpearce** [comment]
Hi @millercentral and @timnovinger, I am sorry for the confusion.  When I originally replied to this thread, I didn't realize you were looking at the wrong repo, I thought I changed this repo to private.

@millercentral, there is a way we can get status from the fan if you use auto or eco but I never used those features myself so it wasn't a priority for me.  

@timnovinger, let me know once you've moved over to the HACS repo and I will close this repo off.

**Comment by timnovinger** [comment]
@davecpearce I got it working after spending a good chunk of the day updating HA to use the new HACS. You sir, are brilliant. It's even supported by the HASS HomeKit bridge and it presents itself as a variable fan. Having the ability to automate it AND control it from any of my Apple devices is PHENOMENAL. Thank you so much!

**Comment by davecpearce** [comment]
Thanks @timnovinger for the kinds words!

**Comment by millercentral** [comment]
Yeah, after installing your component I stopped using auto or eco too, so not really a concern for me anyome either. Thank you again for your work on this!
